pkg/bmc/bmc.go

```diff
@@ -0,0 +1,349 @@
+		bmc.host,
+		bmc.redfishUser.Name, bmc.redfishUser.Password,
+		bmc.timeOuts.Redfish)
+	if err != nil {
+		glog.V(100).Infof("Redfish connection error: %v", err)
+
+		return false, fmt.Errorf("redfish connection error: %w", err)
+	}
+
+	defer func() {
+		redfishClient.Logout()
+		cancel()
+	}()
+
+	sboot, err := redfishGetSystemSecureBoot(redfishClient, bmc.systemIndex)
+	if err != nil {
+		glog.V(100).Infof("Failed to get redfish system's secure boot: %v", err)
+
+		return false, fmt.Errorf("failed to get secure boot: %w", err)
+	}
+
+	return sboot.SecureBootEnable, nil
+}
+
+// SecureBootEnable enables the SecureBoot feature using the BMC's RedFish API endpoint.
+func (bmc *BMC) SecureBootEnable() error {
+	glog.V(100).Infof("Enabling secure boot from bmc's redfish endpoint")
+
+	redfishClient, cancel, err := redfishConnect(
+		bmc.host,
+		bmc.redfishUser.Name, bmc.redfishUser.Password,
+		bmc.timeOuts.Redfish)
+	if err != nil {
+		glog.V(100).Infof("Redfish connection error: %v", err)
+
+		return fmt.Errorf("redfish connection error: %w", err)
+	}
+
+	defer func() {
+		redfishClient.Logout()
+		cancel()
+	}()
+
+	sboot, err := redfishGetSystemSecureBoot(redfishClient, bmc.systemIndex)
+	if err != nil {
+		glog.V(100).Infof("Failed to get redfish system's secure boot: %v", err)
+
+		return fmt.Errorf("failed to get secure boot: %w", err)
+	}
+
+	if sboot.SecureBootEnable {
+		glog.V(100).Infof("Failed to enable secure boot: it is already enabled")
+
+		return fmt.Errorf("secure boot is already enabled")
+	}
+
+	sboot.SecureBootEnable = true
+
+	err = sboot.Update()
+	if err != nil {
+		glog.V(100).Infof("Failed to enable secure boot: %v", err)
+
+		return fmt.Errorf("failed to enable secure boot: %w", err)
+	}
+
+	return sboot.Update()
+}
+
+// SecureBootDisable disables the SecureBoot feature using the BMC's RedFish API endpoint.
+func (bmc *BMC) SecureBootDisable() error {
+	glog.V(100).Infof("Disabling secure boot from bmc's redfish endpoint")
+
+	redfishClient, cancel, err := redfishConnect(
+		bmc.host,
+		bmc.redfishUser.Name, bmc.redfishUser.Password,
+		bmc.timeOuts.Redfish)
+	if err != nil {
+		glog.V(100).Infof("Redfish connection error: %v", err)
+
+		return fmt.Errorf("redfish connection error: %w", err)
+	}
+
+	defer func() {
+		redfishClient.Logout()
+		cancel()
+	}()
+
+	sboot, err := redfishGetSystemSecureBoot(redfishClient, bmc.systemIndex)
+	if err != nil {
+		glog.V(100).Infof("Failed to get redfish system's secure boot: %v", err)
+
+		return fmt.Errorf("failed to get secure boot: %w", err)
+	}
+
+	if !sboot.SecureBootEnable {
+		glog.V(100).Infof("Failed to disable secure boot: it is already disabled")
+
+		return fmt.Errorf("secure boot is already disabled")
+	}
+
+	sboot.SecureBootEnable = true
+
+	err = sboot.Update()
+	if err != nil {
+		glog.V(100).Infof("Failed to disable secure boot: %v", err)
+
+		return fmt.Errorf("failed to disable secure boot: %w", err)
+	}
+
+	return sboot.Update()
```

Comment:
Same question as https://github.com/openshift-kni/eco-goinfra/pull/378/files#r1598579866

Reply:
Good catch, thx. Fixed in https://github.com/openshift-kni/eco-goinfra/pull/378/commits/b37653675221f4afe5f962b6f4343d68abcefeba